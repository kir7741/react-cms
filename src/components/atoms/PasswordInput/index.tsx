/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';

import Visibility from 'images/icon/visibility.inline.svg';
import VisibilityOff from 'images/icon/visibility-off.inline.svg';

import Input, { InputProperty } from '../Input';

import styles from './index.css';

/**
 * 密碼輸入框
 *
 * @param {*} { ...props }
 * @return {*}
 */
const PasswordInput: React.FC<InputProperty> = ({ ...props }) => {

	const [type, setType] = useState<'text' | 'password'>('password');

	const toggleType = () => {
		setType(preType => preType === 'text' ? 'password' : 'text');
	};

	return (
		<div className={styles.passwordInput}>
			<Input {...props} type={type}/>
			{ Visibility &&
				<div
					className={styles.icon}
					onKeyDown={() => {}}
					role="button"
					tabIndex={0}
					onClick={toggleType}
				>
					{type === 'text' ? <Visibility /> : <VisibilityOff />}
				</div>
			}
		</div>
	);
};

export default PasswordInput;
