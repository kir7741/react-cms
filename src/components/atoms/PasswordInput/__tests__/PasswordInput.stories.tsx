import React from 'react';
import { storiesOf } from '@storybook/react';

import PasswordInput from 'components/atoms/PasswordInput';

const stories = storiesOf('atoms/PasswordInput', module);

stories.add('__interactive', () => <PasswordInput />);
