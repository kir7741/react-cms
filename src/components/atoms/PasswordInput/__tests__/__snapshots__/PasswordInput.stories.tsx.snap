// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="passwordInput"
  >
    <div
      class="inputField full"
    >
      <div
        class="inputWrapper"
      >
        <input
          class="input"
          placeholder=""
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="icon"
      role="button"
      tabindex="0"
    >
      <test-file-stub />
    </div>
  </div>
</DocumentFragment>
`;
